src/components/modules/document/common/Upload/modules/Video/modules/Recorder/Recorder.view.vue

```diff
@@ -0,0 +1,185 @@
+<template>
+  <div class="recorder-view">
+    <div class="recorder-view-container">
+      <div>
+        <button
+          class="recorder-view-button-back"
+          @click="onBack"
+        >
+          <v-svg-icon
+            name="arrow-left"
+            width="23px"
+            height="17px"
+          />
+        </button>
+        <div class="recorder-view-title">
+          {{ $t('components.uploadVideo.recorder.title') }}
+        </div>
+      </div>
+      <div class="recorder-view-video">
+        <video-stream
+          :is-recording="isRecording"
+          :is-playing.sync="isPlaying"
+          :file.sync="file"
+          :is-timer-started.sync="isTimerStarted"
+          class="recorder-video-stream"
+        />
+      </div>
+      <div class="recorder-view-controls">
+        <recorder-controls
+          :is-play-available="isPlayAvailable"
+          @confirm="onConfirm"
+          @retake="onRecord"
+        >
+          <record-button
+            :is-recording="isRecording"
+            :is-playing="isPlaying"
+            :is-play-available="isPlayAvailable"
+            :seconds-left="secondsLeft"
+            :seconds-total="MAX_DURATION_SEC"
+            @record="onRecord"
+            @play="onPlay"
+          />
+        </recorder-controls>
+        <count-down-timer
+          :duration="TOTAL_DURATION"
+          :is-locked="isTimerStarted"
+          :counter.sync="secondsLeft"
+          @done="onRecordEnd"
+        />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { ref, computed } from '@vue/composition-api';
+import VSvgIcon from '@endpass/ui/kit/VSvgIcon';
+import RecorderControls from './modules/RecorderControls';
+import CountDownTimer from '@/components/common/CountDownTimer';
+import VideoStream from './modules/VideoStream';
+import RecordButton from './modules/RecordButton';
+
+const MAX_DURATION_SEC = 5;
+const TOTAL_DURATION = MAX_DURATION_SEC * 1000;
+
+export default {
+  name: 'RecorderView',
+
+  setup() {
+    const data = {
```

Comment:
we should not use data, only const. We need `data` section only want to update via complex `updateState` method

Reply:
what about types?